### PR TITLE
Various comment updates and little fixes

### PR DESCRIPTION
**Pull request by matanlurey**

### CODE REVIEW

File: lib/src/backends/chroot/chroot_file_system.dart [hunk 1]
```diff
@@ -62,19 +62,24 @@ class ChrootFileSystem extends FileSystem {
...
-  /// Sets the current working directory for this file system. Note that this
-  /// does *not* proxy to the underlying file system's current directory in
-  /// any way; the state of this file system's current directory is local to
-  /// this file system.
+  /// Sets the current working directory to the specified [path].
+  ///
+  /// **NOTE** This does _not_ proxy to the underlying file system's current
+  /// directory in any way; the state of this file system's current directory
+  /// is local to this file system.
+  /// Gets the path context for this file system given the current working dir.
```
Comment (tvolkert) [inline lib/src/backends/chroot/chroot_file_system.dart]: Looks like this got moved accidentally, such that the docs for `set currentDirectory` are going to apply to this instead.

File: lib/src/backends/chroot/chroot_file_system.dart [hunk 2]
```diff
@@ -205,20 +214,18 @@ class ChrootFileSystem extends FileSystem {
...
   }
 
-  /// Resolves symbolic links on [path] and returns the resulting resolved
-  /// path. The return value will always be an absolute path; if [path] is
-  /// relative, it will be interpreted relative to [from] (or
-  /// [currentDirectory] if [from] is null).
+  /// Resolves symbolic links on [path] and returns the resulting resolved path.
+  ///
+  /// The return value will always be an absolute path; if [path] is relative
+  /// it will be interpreted to [from] (or [currentDirectory] if `null`).
```
Comment (tvolkert) [inline lib/src/backends/chroot/chroot_file_system.dart]: "interpreted relative to"

File: lib/src/backends/chroot/chroot_file_system.dart [hunk 3]
```diff
@@ -318,34 +325,39 @@ class ChrootFileSystem extends FileSystem {
...
+/// What to do when `NOT_FOUND` paths are encountered while resolving.
 enum _NotFoundBehavior {
   ALLOW,
   ALLOW_AT_TAIL,
   THROW,
   MKDIR,
 }
 
-/// File stat representing a not found entity.
+/// A [FileState] representing a `NOT_FOUND` entity.
```
Comment (tvolkert) [inline lib/src/backends/chroot/chroot_file_system.dart]: FileStat

File: lib/src/backends/chroot/chroot_file_system.dart (hunk 1 above)
Comment (matanlurey) [inline lib/src/backends/chroot/chroot_file_system.dart]: Thanks, done.

File: lib/src/backends/chroot/chroot_file_system.dart (hunk 2 above)
Comment (matanlurey) [inline lib/src/backends/chroot/chroot_file_system.dart]: Done.

File: lib/src/backends/chroot/chroot_file_system.dart (hunk 3 above)
Comment (matanlurey) [inline lib/src/backends/chroot/chroot_file_system.dart]: Done.

**Pull request merged**